cedar-policy-core/src/parser/err.rs

```diff
@@ -670,52 +676,61 @@ pub fn expected_to_string(expected: &[String], config: &ExpectedTokenConfig) ->
 // CAUTION: this type is publicly exported in `cedar-policy`.
 // Don't make fields `pub`, don't make breaking changes, and use caution when
 // adding public methods.
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
-pub struct ParseErrors(pub Vec<ParseError>);
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ParseErrors(NonEmpty<ParseError>);
 
 impl ParseErrors {
-    const DESCRIPTION_IF_EMPTY: &'static str = "unknown parse error";
+    /// Construct a `ParseErrors` with a single element
+    pub(crate) fn singleton(err: impl Into<ParseError>) -> Self {
+        Self(NonEmpty::singleton(err.into()))
+    }
+
+    /// Construct a new `ParseErrors` with at least one element
+    pub(crate) fn new(first: ParseError, rest: impl IntoIterator<Item = ParseError>) -> Self {
+        let mut nv = NonEmpty::singleton(first);
+        let mut v = rest.into_iter().collect::<Vec<_>>();
+        nv.append(&mut v);
+        Self(nv)
+    }
 
-    /// Constructs a new, empty `ParseErrors`.
-    pub fn new() -> Self {
-        ParseErrors(Vec::new())
+    /// Construct a new `ParseErrors` from another `NonEmpty` type
+    pub(crate) fn new_from_nonempty(errs: NonEmpty<ParseError>) -> Self {
+        let (first, rest) = errs.split_first();
+        let mut nv = NonEmpty::singleton(first.clone());
+        let mut v = rest.to_vec();
+        nv.append(&mut v);
+        Self(nv)
```

Comment:
why not just `Self(errs)` and avoid the cloning and allocation?

Reply:
Great question. No idea why I did that - will fix in the next commit.